feature: refactored reports module for Vue application

sankha555 opened the pull request:
* [x] partially fixes #232 
* [x] removed HTML code fragments from reports module to make it compatible for JSON serialization for the Vue application

sankha555 commented:
@matthiaskoenig I have refactored the following files to replace the HTML fragments with the dictionary representation of the objects:
- ```sbmlinfo.py```
- ```formating.py```

Tests might fail for now due the changes. Please take a look at the updates and let me know what improvements should  be made. I can then add some appropriate tests too to unit test the new format. Thanks!

matthiaskoenig commented on src/sbmlutils/report/sbmlinfo.py:
```diff
@@ -610,28 +587,29 @@ def info_events(self) -> List[Dict[str, Any]]:
             info["priority"] = priority
 
-            delay = empty_html()
+            delay = ""
```
This should not be empty strings, but None !.
Also use the following pattern consistently for attributes
```
info["delay"] = event.getDelay() if event.isSetDelay() else None

```

matthiaskoenig commented on src/sbmlutils/report/sbmlinfo.py:
```diff
@@ -610,28 +587,29 @@ def info_events(self) -> List[Dict[str, Any]]:
+                "persistent": trigger.persistent
+            }
+
+            priority = ""
```
```
info["priority"] = event.getPriority() if event.isSetPriority() else None
```

matthiaskoenig commented on src/sbmlutils/report/formating.py:
```diff
@@ -7,6 +7,8 @@
+from typing import Dict, List
+
 
 def annotation_to_html(item: libsbml.SBase) -> str:
```
There should be no `*_to_html` method any more ?

matthiaskoenig commented on src/sbmlutils/report/formating.py:
```diff
@@ -36,6 +38,37 @@ def annotation_to_html(item: libsbml.SBase) -> str:
+        info = {}
+        cv = item.getCVTerm(kcv)
+        q_type = cv.getQualifierType()
+        if q_type == 0:
```
check against correct type code from libsbml not int

matthiaskoenig commented on src/sbmlutils/report/formating.py:
```diff
@@ -36,6 +38,37 @@ def annotation_to_html(item: libsbml.SBase) -> str:
+        q_type = cv.getQualifierType()
+        if q_type == 0:
+            qualifier = miriam.ModelQualifierType[cv.getModelQualifierType()]
+        elif q_type == 1:
```
check against correct type code

matthiaskoenig commented on src/sbmlutils/report/formating.py:
```diff
@@ -206,30 +237,32 @@ def modelHistoryToString(mhistory: libsbml.ModelHistory) -> str:
         c = mhistory.getCreator(kc)
         if c.isSetGivenName():
-            cdata.append(c.getGivenName())
+            cdata["given_namw"] = c.getGivenName()
```
typo

matthiaskoenig commented on src/sbmlutils/report/formating.py:
```diff
@@ -206,30 +237,32 @@ def modelHistoryToString(mhistory: libsbml.ModelHistory) -> str:
-        cdata = []
+        cdata = {}
         c = mhistory.getCreator(kc)
         if c.isSetGivenName():
```
use pattern
```
cdata["given_name"] = c.getGivenName() if c.isSetGivenName() else None
```
I.e. use oneliners and set None as default so the fields exist.

matthiaskoenig commented on src/sbmlutils/report/formating.py:
```diff
@@ -358,50 +468,60 @@ def notes(item: libsbml.SBase) -> str:
     :param item: SBML object for which cvterm data has to be displayed
     :return: HTML code fragment enclosing cvterm data for the item
     """
     if item.isSetAnnotation():
```
Makes more sense to have this check in `annotation_dict`

matthiaskoenig reviewed (commented):
Most of the things are okay. This is difficult to review and we have to probably make a code review together on this. I suggest you finish this up as well as the JSON serialization and then we would refactor the code together in a session.

sankha555 commented:
@matthiaskoenig Sure Sir, I will resolve the suggested reviews first and then move to the JSON serialization. We can then have a session on reviewing the code changes, and then I can move on to the next module. Thanks for the reviews!

sankha555 commented:
@matthiaskoenig I have added the suggested changes and the serialization. I have also added a small test to print the json to a file, but unfortunately it is not returning any results. I had tried to print dummy results as well but the functions in ```sbmlreport.py``` do not seem to be returning any value (everytime ```None``` is returned, even if I try to return a simple string like "Hello"). 

Could you please take a look at the changes and also suggest what might be perhaps causing the above issue? Please let me know if you need some more information from me. I can also be available for reviewing the code with you at your time and convenience. Thanks!
